Script to run AUCell on all samples using EWS-FLI1 high/low gene signatures

allyhawkins opened the pull request:
### Purpose/implementation Section

#### Please link to the GitHub issue that this pull request addresses.

Closes #985 

#### What is the goal of this pull request?

Here I am adding two new scripts, one to run `AUCell` on a single SCE object using a set of custom gene signatures for defining tumor cell states (mainly EWS-FLI1 high and EWS-FLI1 low) and a second script to run the first script on all samples in the Ewing project. Ultimately, we want to use the results from AUCell to help label cells based on the specific cell state they are in. 

I am using the two custom gene sets that we have stored in `references/gene_signatures` that are marker gene lists for EWS-FLI1 high and low along with a [set of MSigDB gene sets](https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses/cell-type-ewings/references#gene-signatures) that we have identified from the literature as being potentially useful. It's pretty quick to run `AUCell` so I figured it wouldn't hurt to just use all the gene sets in that list. 

I did not include genes in our two marker gene lists, `visser-all-marker-genes.tsv` and `tumor-cell-state-markers.tsv` since the gene lists for each cell type there are quite small and would probably skew the AUC results. 

#### Briefly describe the general approach you took to achieve this goal.

- Although we previously had a [script for AUCell](https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/main/analyses/cell-type-ewings/scripts/auc-singler-workflow/01-run-aucell.R), it was for a pretty specific use case where we are defining tumor cells. I didn't want to have to make huge edits in that workflow and the code was different enough that I chose to make a new script for this. 
- This script takes in a single SCE object, the custom gene sets, and a percentage to use for determining the `aucMaxRank`, which I set to the default of 0.01. 
- The output of this script is a TSV file with AUC values for each cell and each gene set. I also chose to include the AUC threshold value that is reported by `AUCell`. I don't know that we will use it, but I think it could be helpful when we go to plot this data. 
- I then wrote a wrapper script that runs AUCell using these gene signatures on every sample in the project and saves each TSV to the results directory. 
- The last thing I did was update the documentation where necessary. I know this technically isn't a new "workflow", since we're just running one script, but I still followed a similar format in how I documented it and included it in the main README. Once we create our notebooks that go sample by sample, I would like to reorganize some of the things in this module and move things that aren't used anymore to a sub folder so that things are less crowded and easier to understand. But for now, I just added it to existing documentation. 


#### If known, do you anticipate filing additional pull requests to complete this analysis module?

Yes see #993 

### Results

#### What is the name of your results bucket on S3?

`s3://researcher-211125375652-us-east-2/cell-type-ewings/results/aucell-ews-signatures`

#### What types of results does your code produce (e.g., table, figure)?

TSV files with the AUC values

#### What is your summary of the results?

Coming next! 


### Author checklists

_Check all those that apply._
_Note that you may find it easier to check off these items after the pull request is actually filed._


#### Analysis module and review

- [ ] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [x] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [x] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [ ] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

sjspielman commented on analyses/cell-type-ewings/exploratory_analysis/07-identify-max-rank-aucell.Rmd:
```diff
@@ -64,7 +64,7 @@ sce_list <- sce_list |>
     
     genes_to_remove <- rowData(sce)$detected > 0 
-    filtered_sce <- sce[genes_to_remove > 0 , ]
+    filtered_sce <- sce[genes_to_remove , ]
```
probably re-knit this too?

sjspielman commented on analyses/cell-type-ewings/scripts/aucell-ews-signatures/01-aucell.R:
```diff
@@ -0,0 +1,192 @@
+auc_results <- AUCell::AUCell_run(
+  counts_mtx, 
+  collection, 
+  aucMaxRank = max_rank,
+  #BPPARAM = bp_param
```
Is this meant to be commented out?

sjspielman commented on analyses/cell-type-ewings/scripts/aucell-ews-signatures/01-aucell.R:
```diff
@@ -0,0 +1,192 @@
+      unique()
+  })
+
+# grab msigdb gene sets 
+ews_gene_sets <- c(
+  "staege" = "STAEGE_EWING_FAMILY_TUMOR",
+  "miyagawa_up" = "MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_UP",
+  "miyagawa_down" = "MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_DN",
+  "zhang"= "ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION", 
+  "riggi_up" = "RIGGI_EWING_SARCOMA_PROGENITOR_UP",
+  "riggi_down" = "RIGGI_EWING_SARCOMA_PROGENITOR_DN",
+  "kinsey_up" = "KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_UP",
+  "kinsey_down"= "KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_DN"
+)
```
I'm actually wondering now whether we want to make this a stand-alone TSV file to read in here, and we can format it into a named vector in the script since I think that may be more maintainable should anything change here.

I'm also wondering about the Wrenn msigdb sets as seen in `references/README.md`.. Do we want to use those here?
 
```
- [`HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION.html)
- [`REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/GOBP_REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION.html)
```

sjspielman reviewed (commented):
This all looks good to me, and I was able to run the script locally. Left some comments but nothing major, let me know if I can clarify!

allyhawkins commented:
> I'm actually wondering now whether we want to make this a stand-alone TSV file to read in here, and we can format it into a named vector in the script since I think that may be more maintainable should anything change here.
> 
> I'm also wondering about the Wrenn msigdb sets as seen in `references/README.md`.. Do we want to use those here?
> 
> ```
> - [`HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION.html)
> - [`REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION`](https://www.gsea-msigdb.org/gsea/msigdb/human/geneset/GOBP_REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION.html)
> ```

I updated this to include a TSV file `msigdb-gene-sets.tsv` that gets read in and used to grab the gene sets from MSigDB. I also added those two extra gene sets. To be honest, I just didn't want to deal with getting gene sets from different collections, but that's silly and we should include them which is easier to do with the TSV file! This involved some code updates to account for the fact that we have multiple collections, but nothing too crazy. 

I re-ran the workflow and updated the results on S3 to reflect these changes. This should be ready for another look!

sjspielman commented on analyses/cell-type-ewings/run-aucell-ews-signatures.sh:
```diff
@@ -0,0 +1,75 @@
+# This script is used to get AUC values for a set of EWS-FLI1 high/low gene signatures using AUCell
+# The input gene signatures are those saved in `references/gene_signatures` and the following MSigDb gene sets: 
+
+# `STAEGE_EWING_FAMILY_TUMOR`
+# `MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_UP`
+# `MIYAGAWA_TARGETS_OF_EWSR1_ETS_FUSIONS_DN`
+# `ZHANG_TARGETS_OF_EWSR1_FLI1_FUSION`
+# `RIGGI_EWING_SARCOMA_PROGENITOR_UP`
+# `RIGGI_EWING_SARCOMA_PROGENITOR_DN`
+# `KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_UP`
+# `KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_DN`
```
I think you can remove this now since we have the TSV file, but the comment above might need to be updated a bit about where the reference files are?

sjspielman commented on analyses/cell-type-ewings/scripts/aucell-ews-signatures/01-aucell.R:
```diff
@@ -0,0 +1,214 @@
+
+# make sure input files exist
+stopifnot(
+  "sce file does not exist" = file.exists(opt$sce_file),
```
This will fail with `Error in file.exists(opt$sce_file) : invalid 'file' argument` if an SCE file is not provided in the first place, so you might want to first have a separate check that it's not null.

sjspielman commented on analyses/cell-type-ewings/references/msigdb-gene-sets.tsv:
```diff
@@ -0,0 +1,11 @@
+riggi_down	RIGGI_EWING_SARCOMA_PROGENITOR_DN	C2	CGP
+kinsey_up	KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_UP	C2	CGP
+kinsey_down	KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_DN	C2	CGP
+hallmark_EMT	HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION	H	
```
Can we actually add `NA` for the subcategory here instead of leaving empty? It does get read as `NA` within R, but seems safer to actually write it out here.

allyhawkins replied:
Yes, I'll change this but then will still keep the change to use NULL in the script.

sjspielman commented on analyses/cell-type-ewings/references/msigdb-gene-sets.tsv:
```diff
@@ -0,0 +1,11 @@
+kinsey_up	KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_UP	C2	CGP
+kinsey_down	KINSEY_TARGETS_OF_EWSR1_FLII_FUSION_DN	C2	CGP
+hallmark_EMT	HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION	H	
+gobp_ECM	REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION	C5	BP
```
I noticed that none of the result TSVs were associated with this gene set, and when I poked around more with the code, indeed it seems anything that gets returned with `C5` & `BP` gets filtered out because the gene set name is not actually present in MSigDB output. 

But, good news, it's just a typo! https://www.gsea-msigdb.org/gsea/msigdb/cards/GOBP_REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION. I replaced this term in the TSV file then there are indeed results.

So, this should be replaced and the code should be re-run. This may also need to be updated in READMEs.

```suggestion
gobp_ECM	GOBP_REGULATION_OF_EXTRACELLULAR_MATRIX_ORGANIZATION	C5	BP
```

sjspielman commented on analyses/cell-type-ewings/run-aucell-ews-signatures.sh:
```diff
@@ -0,0 +1,75 @@
+
+# gene signature directory 
+gene_signatures_dir="${module_dir}/references/gene_signatures"
+msigdb_geneset_file="${module_dir}/references/msigdb-gene-sets.txt"
```
```suggestion
msigdb_geneset_file="${module_dir}/references/msigdb-gene-sets.tsv"
```

with this fix, all the code runs for me!

allyhawkins replied:
😬 co-pilot filled this in for me while I was typing and I was like ohhh it got it right... nope...

sjspielman reviewed (approved):
I caught a few bugs which need to be fixed and the code re-run accordingly, but otherwise I think this is good to go! I don't need to see this again unless you'd like to tag me back.

allyhawkins commented:
Noting that made the updates requested, re-ran the script, and then re-uploaded results. I'll merge this in once checks pass!
